nibabel/nifti1.py

```diff
@@ -379,13 +381,107 @@ def write_to(self, fileobj, byteswap):
         fileobj.write(b'\x00' * (extstart + rawsize - fileobj.tell()))
 
 
+class Nifti1DicomExtension(Nifti1Extension):
+    """NIfTI1 DICOM header extension
+
+    This class is a thin wrapper around pydicom to read a binary DICOM
+    byte string. If pydicom is not available, it silently falls back to the
+    standard NiftiExtension class.
+    """
+    def __init__(self, code, content, parent_hdr=None):
+        """
+        Parameters
+        ----------
+        code : int|str
+          Canonical extension code as defined in the NIfTI standard, given
+          either as integer or corresponding label
+          (see :data:`~nibabel.nifti1.extension_codes`)
+        content : bytes|pydicom Dataset|None
+          Extension content - either a bytestring as read from the NIfTI file
+          header or an existing pydicom Dataset. If a bystestring, the content
+          is converted into a Dataset on initialization. If None, a new empty
+          Dataset is created.
+        parent_hdr : :data:`~nibabel.nifti1.Nifti1Header`
+          Optional - If a dicom extension belongs to an existing
+          :data:`~nibabel.nifti1.Nifti1Header`, it may be provided here to
+          ensure that the DICOM dataset is written with correctly corresponding
+          endianness; otherwise it is assumed the dataset is little endian.
+
+        Notes
+        -----
+
+        code should always be 2 for DICOM.
+        """
+
+        self._code = code
+        if parent_hdr:
+            self._is_little_endian = parent_hdr.endianness == '<'
+        else:
+            self._is_little_endian = True
+        if isinstance(content, pdcm.dataset.Dataset):
+            self._is_implicit_VR = False
+            self._raw_content = self._mangle(content)
+            self._content = content
+        elif isinstance(content, bytes):  # Got a byte string - unmangle it
+            self._raw_content = content
+            self._is_implicit_VR = self._guess_implicit_VR()
+            ds = self._unmangle_and_verify(content, self._is_implicit_VR,
+                                           self._is_little_endian)
+            self._content = ds
+        elif content == None:  # Otherwise, initialize a new dicom dataset
+            self._is_implicit_VR = False
+            self._content = pdcm.dataset.Dataset()
+        else:
+            raise TypeError("content must be either a bytestring or a pydicom "
+                            "Dataset. Got %s" % content.__class__)
+
+    def _unmangle_and_verify(self, content, is_implicit_VR, is_little_endian):
+        """"Decode and verify dicom dataset"""
+        ds = self._unmangle(content, is_implicit_VR, is_little_endian)
+        for elem in ds:
+            if elem.VR not in pdcm.values.converters.keys():
+                raise StandardError  # Ensure that all VRs are valid
```

Comment:
Actually, the only case I can find that will cause this is to give a malformed implicit VR, which is correctly reported by pydicom as a KeyError (`KeyError: "Unknown DICOM tag (0054, 0099) - can't look up VR"`) and not caught as a StandardError. So, I'm going to remove this validation code completely.